### PR TITLE
Feat / Account keys UI

### LINKED ISSUE #1835: [settings] account keys form UI
## Description

Clicking the "Edit details" button on the settings UI opens a modal that allows to enter the account keys and the webhook secret.
Clicking "Enter account keys (advanced)" on the onboarding UI opens a modal that allows to enter the **live** account keys and the webhook secret (PLEASE NOTE: the latest designs have been updated, but let's use the same modal for now, as a first step).
Clicking "enable test mode" without test keys should open _the same form_ (without the live/test keys tabs) that allows to enter the test keys (in a different modal).

The modal persists the data to the backend (use `wp/data` implementation from https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2004 ) immediately after confirmation.

live keys modal

Keys edit modal

The "Test connection" functionality is implemented separately: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1806

## Acceptance criteria
- Enabling "Enable test mode" without test keys opens the modal that allows to enter test keys
- Disabling "Enable test mode" without live keys opens the modal that allows to enter live keys
- Clicking the "Edit details" button opens the modal
- Clicking "Enter account keys (advanced)" on the onboarding UI opens the modal
- Modal matches the latest design (please be aware that the attached screenshot might not be updated)
- Modal is dismissable
- Clicking the "Save & connect" button persists the data to the backend
- Once the data is successfully persisted, the modal is closed
- dismissing the modal should not persist the data, and reopening it should show the persisted data (not the previously edited one)
- On the onboarding UI, the page is refreshed once the data is successfully persisted

### PR DESCRIPTION
**Pull request by tommyshellberg**
# Changes proposed in this Pull Request:

Issue: Add Account Keys UI to the new settings area.
Fixes #1835 

For acceptance criteria take a look at #1835 

Description: Create a pop-up modal which allows for adding Account Keys using the new react-based settings UI.
This was started before the team Heisenberg meetup and is functional but due to meetup planning responsibilities I couldn't devote the time needed to get it 100%. There are some failing tests still and it could use some optimization and polish. Marking as Draft for this to get picked up since the team will be focused as of now on WooPay.

# Testing instructions

There are three ways to open the modal:

1. Click **Edit Account Keys** within the General settings area on the settings panel of the react-based settings area.
2. Click **Enter account keys (advanced)** within the onboarding UI(this requires deleting your Stripe keys to prompt onboarding).
3. Click **Edit Account Keys** drop-down menu link within the **Account Details** area within the **Settings** panel.

Once the modal is open you should be able to switch between Test & Live mode; the keys shown there should be loaded from your saved Stripe keys. 
You should be able to save the keys and receive a toast notification (success or fail). 

There are some tests to make sure the modal opens but I think a test to ensure that upon form submission the `saveAccountKeys()` function is called with the correct arguments would be beneficial.

There is some prop drilling at the moment and perhaps an optimization could be made to create a custom react hook for opening the modal rather than passing down props:

https://github.com/woocommerce/woocommerce-gateway-stripe/compare/feat/account-keys-ui?expand=1#diff-d85d448546d4ee9c6d71de46179d526555442e9b6e0aa03f844d6bcb79b8e6a6R55

### CODE REVIEW

File: client/settings/payment-settings/account-keys-modal.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+				onSelect={ onTabSelect }
+				tabs={ [
+					{
+						name: 'live',
+						title: 'Live',
+						className: 'live-tab',
+					},
+					{
+						name: 'test',
+						title: 'Test',
```
Comment (ricardo) [inline client/settings/payment-settings/account-keys-modal.js]: Should these (title) be translatable strings?

**Comment by ricardo** [comment]
Looks great and tests well.

From the user's perspective, I personally had a small issue with the UX:

- After I cleared all my keys to test the onboarding modal, I tried to input both the live and test keys at the same time, and then hit "Save changes" at the end. It ended up not saving both keys, but only the information from the tab I selected last.

This might be a common behavior for some users.

Perhaps we could fix that by saving all keys upon clicking "Save changes", similar to the current (old settings) behavior, or tweaking the messaging in a way that it's clear that only the current tab will be saved - perhaps with the button "Save test keys".

---

Another tiny 🤏 detail I noticed: Maybe this text wasn't thought for when you're just editing the keys and not adding them for the first time:

> To enable the live mode, get the account keys from your Stripe Account (we'll save them for you **so you won't have to do this every time**).

The part "so you won't have to do this every time" kind of assumes it's the first time I'm opening up the modal and adding/editing the keys. This one is not a big deal though. I don't really have any good suggestions for tweaks right now. Maybe we should only display this text if the keys are blank. 🤔

**Review by ricardo — APPROVED** [review]
LGTM and tests well - pre-approving.

I only added one minor comment regarding the changes, but I left a separate comment above regarding a feedback on the UX.

Let me know if that makes sense or if we don't want to tweak the UX for this PR.

**Comment by diegocurbelo** [comment]
> After I cleared all my keys to test the onboarding modal, I tried to input both the live and test keys at the same time, and then hit "Save changes" at the end. It ended up not saving both keys, but only the information from the tab I selected last.

As you suggested @ricardo, I updated the `save` button text to `Save live keys` and `Save test keys` to be explicit about the behavior.

> To enable the live mode, get the account keys from your Stripe Account (we'll save them for you so you won't have to do this every time).
The part "so you won't have to do this every time" kind of assumes it's the first time I'm opening up the modal and adding/editing the keys. This one is not a big deal though. I don't really have any good suggestions for tweaks right now. Maybe we should only display this text if the keys are blank. 🤔

Agree... removed that text

**Review by ricardo — APPROVED** [review]
LGTM

**Pull request merged**